docs/contributor-docs/global-style-utils.md

```diff
@@ -0,0 +1,201 @@
+
+### SharedTokens Integration
+
+```ts
+---
+type: code
+---
+const generateStyle = (
+  componentTheme: ComponentTheme,
+  params: StyleParams,
+  sharedTokens: SharedTokens // ← Automatically injected by useStyle
+) => {
+  return {
+    myComponent: {
+      // Use sharedTokens for margin calculations
+      margin: calcMarginFromShorthand(params.margin, sharedTokens.margin),
+
+      // Use sharedTokens for focus outline styles
+      ...calcFocusOutlineStyles(sharedTokens.focusOutline, {
+        focusColor: 'info'
+      })
+    }
+  }
+}
+```
+
+### Available SharedTokens Properties
+
+- **`sharedTokens.margin`**: Spacing token map for use with `calcMarginFromShorthand`
+- **`sharedTokens.focusOutline`**: Focus outline theme configuration for use with `calcFocusOutlineStyles`
+- **Other shared design tokens**: Additional properties available across all components
+
+This integration ensures that all components have consistent access to the design system's shared tokens without needing to manually pass theme data to each utility function.
+
+## calcMarginFromShorthand
+
+The `calcMarginFromShorthand` utility converts shorthand margin values into CSS margin strings using the theme's spacing map.
+
+### Usage
+
+```ts
+---
+type: code
+---
+import { calcMarginFromShorthand } from '@instructure/emotion'
+
+const cssMargin = calcMarginFromShorthand(value, spacingMap)
+```
+
+### Parameters
+
+- **value** (`Spacing | undefined`): A string representing margin values in shorthand format
+- **spacingMap** (`{ [key: string]: any }`): An object mapping spacing token names to CSS values
+
+### Examples
+
+```ts
+---
+type: code
+---
+// Within a generateStyle function (typical usage)
+const generateStyle = (componentTheme, params, sharedTokens) => ({
+  myComponent: {
+    // Single value
+    margin: calcMarginFromShorthand('medium', sharedTokens.margin)
+    // Returns: '1rem'
+  }
+})
+
+// Multiple values (CSS shorthand format)
+calcMarginFromShorthand('small medium large small', sharedTokens.margin)
+// Returns: '0.5rem 1rem 1.5rem 0.5rem'
+
+// Nested theme tokens
+calcMarginFromShorthand('between.cards.md', sharedTokens.margin)
+// Returns: '1rem' (assuming sharedTokens.margin.between.cards.md exists)
+
+// Mixed values
+calcMarginFromShorthand('16px medium auto small', sharedTokens.margin)
+// Returns: '16px 1rem auto 0.5rem'
+```
+
+### How it works
+
+1. **Splits the input**: The function splits the input string by spaces to handle CSS shorthand format (top, right, bottom, left)
+2. **Resolves tokens**: For each value, it checks if it's a theme token path (e.g., 'between.cards.md')
+3. **Maps values**: Theme tokens are resolved using the spacing map, while literal values (like '16px', 'auto') are passed through unchanged
+4. **Joins result**: All resolved values are joined back with spaces to create a valid CSS margin value
```

Comment:
I would remove this and add it to the JSDoc of `calcMarginFromShorthand`. Here just mention what it does with an example